File: RedSheet.DbEntities/Models/vTheirTeamMemberRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vTheirTeamMemberRecords", Schema = "dbo")]
	[Serializable]
	public partial class vTheirTeamMemberRecord  
	{
 
	[MaxLength(1000)]
	public string Description { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string PersonalityColor { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int PersonalityId { get; set; } 
 
	[MaxLength(50)]
	public string PersonalityKey { get; set; } 
 
	[Required] 
	[MaxLength(500)]
	public string Position { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectNegotiationId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TheirTeamMemberId { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string TheirTeamMemberName { get; set; } 

	public vTheirTeamMemberRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-their-team-member-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vTheirTeamMemberRecord {
    constructor(vTheirTeamMemberRecord?: vTheirTeamMemberRecord )  {
        let properties = [ "description", "personalityColor", "personalityId", "personalityKey", "position", "projectNegotiationId", "theirTeamMemberId", "theirTeamMemberName",];
        for (let property of properties)
            if (vTheirTeamMemberRecord && vTheirTeamMemberRecord[property])
                this[property] = vTheirTeamMemberRecord[property];
    }
 
    @maxLength(1000)
	description : string =   undefined;
 
    @required()
    @maxLength(50)
	personalityColor : string =   undefined;
 
	personalityId : number =   0 ;
 
    @maxLength(50)
	personalityKey : string =   undefined;
 
    @required()
    @maxLength(500)
	position : string =   undefined;
 
    @range(1,2147483647)
	projectNegotiationId : number =   undefined;
 
    @range(1,2147483647)
	theirTeamMemberId : number =   undefined;
 
    @required()
    @maxLength(50)
	theirTeamMemberName : string =   undefined;


}
